client/extensions/woocommerce/app/promotions/promotion-form-card.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormTextInput from 'components/forms/form-text-input';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import PriceInput from 'woocommerce/components/price-input';
+import promotionModels, { hasField, isRequiredField } from './promotion-models';
+
+function renderRequiredLabel( isRequired, translate ) {
+	if ( ! isRequired ) {
+		return null;
+	}
+
+	return (
+		<span className="promotions__promotion-form-label-required">
+			{ translate( 'Required' ) }
+		</span>
+	);
+}
+
+function renderCouponCode( siteId, model, promotion, editPromotion, translate ) {
+	if ( ! hasField( model, 'couponCode' ) ) {
+		return null;
+	}
+
+	const onCouponCodeChange = ( e ) => {
+		const couponCode = e.target.value;
+		editPromotion( siteId, promotion, { couponCode, type: promotion.type } );
+	};
+
+	return (
+		<FormFieldset>
+			<FormLabel>{ translate( 'Coupon code' ) }</FormLabel>
+			{ renderRequiredLabel( isRequiredField( model, 'couponCode' ), translate ) }
+			<FormTextInput
+				value={ promotion.couponCode }
+				onChange={ onCouponCodeChange }
+				placeholder={ translate( 'Enter coupon code' ) }
+			/>
+			<div className="promotions__promotion-form-field-footnote">
+				{ translate( 'Only apply this promotion when the customer supplies the coupon code' ) }
+			</div>
+		</FormFieldset>
+	);
+}
+
+function renderAmount( siteId, model, promotion, editPromotion, translate, currency ) {
+	if ( ! hasField( model, 'amount' ) ) {
+		return null;
+	}
+
+	return (
+		<FormFieldset>
+			<FormLabel>{ translate( 'Discount' ) }</FormLabel>
```

Comment:
Maybe add translation context (i.e. "discount the noun, not verb")

Reply:
Added noun context, thanks